### ISSUE TITLE
NPE for some styles

### ISSUE THREAD

**Issue by Siedlerchr** [michel-kraemer/citeproc-java#116]
Originally reported here with some styles https://github.com/JabRef/jabref/issues/8421

I could reproduce it (e.g. Oxford Art Journal). The preview uses the Test Entry:
https://github.com/JabRef/jabref/blob/bb011c9313367a28990ae213b3920fe6cd10d1dc/src/main/java/org/jabref/logic/util/TestEntry.java#L16-L39

```
Error while generating citation style
java.lang.NullPointerException: Cannot invoke "de.undercouch.citeproc.csl.internal.SBibliography.render(de.undercouch.citeproc.csl.internal.RenderContext)" because the return value of "de.undercouch.citeproc.csl.internal.SStyle.getBibliography()" is null
	at org.jabref.merged.module@100.0.0/de.undercouch.citeproc.CSL.makeBibliography(Unknown Source)
	at org.jabref.merged.module@100.0.0/de.undercouch.citeproc.CSL.makeBibliography(Unknown Source)
	at org.jabref@100.0.0/org.jabref.logic.citationstyle.CSLAdapter.makeBibliography(Unknown Source)
	at org.jabref@100.0.0/org.jabref.logic.citationstyle.CitationStyleGenerator.generateCitations(Unknown Source)
	at org.jabref@100.0.0/org.jabref.logic.citationstyle.CitationStyleGenerator.generateCitation(Unknown Source)
	at org.jabref@100.0.0/org.jabref.logic.citationstyle.CitationStylePreviewLayout.generatePreview(Unknown Source)
	at org.jabref@100.0.0/org.jabref.gui.preview.PreviewViewer.lambda$update$4(Unknown Source)
	at org.jabref@100.0.0/org.jabref.gui.util.BackgroundTask$1.call(Unknown Source)
	at org.jabref@100.0.0/org.jabref.gui.util.DefaultTaskExecutor$1.call(Unknown Source)
	at org.jabref.merged.module@100.0.0/javafx.concurrent.Task$TaskCallable.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

```

**Comment by michel-kraemer** [comment]
Thanks for reporting this! The styles mentioned in the JabRef issue do not contain instructions to format bibliographies. It seems they're only used to format citations.

A NullPointerException is of course not very helpful in this case. What would you suggest instead? IllegalArgumentException? IllegalStateException? Maybe a method to determine if a style can be used to format bibliographies, which you can use upfront to filter out the styles that would not work anyhow? (e.g. `CSL.canFormatBibliographies(style)`, plus maybe `CSL.canFormatCitations(style)`)

**Comment by Siedlerchr** [comment]
Ah thanks for the info, didn't know there was as difference. 
I guess a method like `CSL.canFormatBibliographies(style)` would be ideal, so we can filter them out.  Preferable over an exception.

**Comment by michel-kraemer** [comment]
I've checked the [CSL specification](https://docs.citationstyles.org/en/stable/specification.html) and the `<cs:citation>` element is mandatory, whereas `<cs:bibliography>` is optional. This means it is OK to only add `canFormatBibliographies()`. `canFormatCitations()` is not necessary. I just pushed a commit with this change. I also made sure `makeBibliography` throws an `IllegalStateException` (instead of an NPE) with a sensible message if the style does not support formatting bibliographies.
